internal/tasks/schedulertask.go

```diff
@@ -429,3 +432,64 @@ func (sm *SchedulerTask) scheduleSubscription(ctx context.Context, sub *subscrip
 	logger.Ctx(ctx).Infow("schedulertask: successfully assigned nodebinding for subscription/partition combo", "topic", sub.Topic, "subscription", sub.Name, "partition", partition)
 	return nil
 }
+
+// rebalanceSubs achieves the following in the order outline:
+// 1. Fetch and order nodes in decreasing order of amount of node bindings
+// 2. Rebalance nodebindings from nodes with greater than avg nodebindings
+func (sm *SchedulerTask) rebalanceSubs(ctx context.Context) error {
+	// Fetch the latest node bindings after deletions and schedule missing bindings
+	nodeBindings, err := sm.nodeBindingCore.List(ctx, nodebinding.Prefix)
+	if err != nil {
+		logger.Ctx(ctx).Errorw("Rebalancing: error fetching new node binding list", "error", err)
+		return err
+	}
+	nodeMap := make(map[string][]*nodebinding.Model)
+	nodes := make([]string, 0, len(nodeMap))
+
+	// Create a map with nodebinding list such as { "node1" : ["nodebinding1", "nodebinding2"] }
+	for _, nb := range nodeBindings {
+		if len(nodeMap[nb.NodeID]) != 0 {
+			nodeMap[nb.NodeID] = append(nodeMap[nb.NodeID], nb)
+		} else {
+			nodeMap[nb.NodeID] = []*nodebinding.Model{nb}
+			nodes = append(nodes, nb.NodeID)
+		}
+	}
+
+	// Sort nodes in decreasing order of number of nodebindings
+	sort.SliceStable(nodes, func(i, j int) bool {
+		return len(nodeMap[nodes[i]]) > len(nodeMap[nodes[j]])
+	})
+
+	avgNodebindings := 0
+	if len(nodeMap) != 0 {
+		avgNodebindings = len(nodeBindings) / len(nodeMap)
+	}
+
+	invalidBindings := make(map[string]*nodebinding.Model)
+	for i := 0; i < len(nodes); i++ {
+		if len(nodeMap[nodes[i]]) <= avgNodebindings {
+			continue
+		}
+
+		nodesToRebalance := len(nodeMap[nodes[i]]) - avgNodebindings
```

Comment:
this actually gives extra nodeBindings for that pod that needs to be rebalanced, right..?
Variable name coveys something else

Reply:
updated var name